pkg/source/source.go

```diff
@@ -212,12 +222,11 @@ func (is *Informer) String() string {
 var _ Source = Func(nil)
 
 // Func is a function that implements Source.
-type Func func(context.Context, handler.EventHandler, workqueue.RateLimitingInterface, ...predicate.Predicate) error
+type Func func(context.Context, workqueue.RateLimitingInterface) error
```

Comment:
It does implement source, but does it make sense? There will be no handler available to funnel the events, no place to store it, maybe an example of how this will work now be helpful? Or just remove it

Reply:
> It does implement source, but does it make sense? There will be no handler available to funnel the events, no place to store it, maybe an example of how this will work now be helpful?

As mentioned earlier, there is no need for the concept of a handler when not using a kube watch. A simple example of how to use this would be something like the below, that being said the point of this is to simplify implementing the interface for whatever people want, so if we added the example in code, people might think that is how its supposed to be used but its only one of potentially many ways:
```
queueInitialized := make(chan struct{})
var workqueue workqueue.RateLimitingInterface

ctrl.Watch(source.Func(func(_ context.Context, w workqueue.RateLimitingInterface) error {
     workqueue = w
     close(queueInitialized)
}))

add := func(in types.NamespacedName){
    <- queueInitialized
    workqueue.Add(in)
}
```